### PR TITLE
Move to server generation

### PR DESCRIPTION
**Pull request by tommyjcarpenter**
This is like two weeks of on and off work. 

Originally, I was thinking to do these 3 (sstr, dec, meb) one at a time, but it was actually a lot easier to do them in one shot to get the abstractions right. 

The large amount of code deleted in this PR was already moved to the server side (so the UI can (and already is) use it to)

### CODE REVIEW

File: merqube_client_lib/api_client/base.py [hunk 1]
```diff
@@ -153,7 +154,12 @@ def replace_target_portfolio(self, index_id: str, target_portfolio: dict[str, An
 
         (You can get the index id by doing get_index_manifest(index_name)["id"])
         """
-        return cast(ResponseJson, self.session.put(f"/index/{index_id}/target_portfolio", json=target_portfolio).json())
+
+        for itp in target_portfolio:
```
Comment (tommyjcarpenter) [inline merqube_client_lib/api_client/base.py:158]: @mrondon note: array. UI should take same logic on update (and initial)

File: merqube_client_lib/pydantic_types.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # generated by datamodel-codegen:
 #   filename:  openapi.yaml
-#   timestamp: 2023-08-09T00:09:30+00:00
+#   timestamp: 2023-08-17T17:01:29+00:00
```
Comment (tommyjcarpenter) [inline merqube_client_lib/pydantic_types.py:3]: this file is autogenerated based on indexapi types, you can skip

File: merqube_client_lib/templates/equity_baskets/schema.py [hunk 3]
```diff
@@ -42,25 +33,7 @@ def get_example(cls) -> dict[str, Any]:
                 d[field_name] = model_field.field_info.extra["example"]
 
         instance = cls(**d)
-        return cast(dict[str, Any], json.loads(instance.json()))
-
-    @validator("timezone")
```
Comment (tommyjcarpenter) [inline merqube_client_lib/templates/equity_baskets/schema.py:47]: this is server side

File: merqube_client_lib/templates/equity_baskets/creators.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    ClientSSTRConfig,
+)
+from merqube_client_lib.templates.equity_baskets.util import EquityBasketIndexCreator
+
+logger = get_module_logger(__name__, level=logging.DEBUG)
+
+
+def read_file(filename: str) -> list[Any]:
+    """reads portfolio/overrides files"""
+    return list(pd.read_csv(filename, float_precision="round_trip").to_dict(orient="index").values())
```
Comment (jperozo) [inline merqube_client_lib/templates/equity_baskets/creators.py:26]: This utility its going to read just file from us? If its the case, you can ignore this comment

File: tox.ini [hunk 5]
```diff
@@ -22,7 +22,6 @@ setenv =
 commands_pre =
     poetry install --sync
 commands =
-    poetry run mypy -p merqube_client_lib --strict
```
Comment (jperozo) [inline tox.ini:25]: :eyes:

**Review by jperozo — APPROVED** [review]
Mostly deleting thins. lgtm, tho!

File: tox.ini (hunk 5 above)
Comment (tommyjcarpenter) [reply tox.ini:25]: haha good catch

**Pull request merged**